libs/mpi_base/src/mpi_environment.cpp

```diff
@@ -187,6 +197,10 @@ namespace hpx { namespace util {
 
         this_rank = rank();
 
+#if !defined(HPX_HAVE_NETWORKING)
+        // all ranks are just local nodes when networking is off
+        rtcfg.mode_ = hpx::runtime_mode::local;
+#else
```

Comment:
I don't think this will do what one would expect. This will start only the local runtime. I think actions and components will still need the AGAS etc. stubs even if they don't do anything. Not sure however... I would leave this part unchanged.

Reply:
I rearranged this a bit so that `local` is only used when the distributed runtime is off, `console` when networking is off but distributed runtime is on, and the old code when networking is on.